### ISSUE TITLE
Allow customisation of the RootEndpoint in IPv8 before the REST API starts

### ISSUE THREAD

**Issue by devos50** [Tribler/py-ipv8#833]
I'm currently trying to fix the REST API tests in AnyDex and use the `RESTTestBase` to run the unit tests. Even though this seems to work great within IPv8 itself, and apparently does not require much code, I find that it is cumbersome to customise and re-use this class in a project that uses IPv8 as a dependency.

AnyDex (and Tribler) uses their own `RESTManager` that starts/stops the REST API. Both these projects define their own `RootEndpoint` and then add the IPv8 `RootEndpoint` to the root endpoint manually. I would like to get rid of the `RESTManager` in AnyDex (since it was originally copied code from an older IPv8 version) and use the `RESTManager` provided by IPv8 instead. The problem, however, is that IPv8 does not give me any control over the `RootEndpoint` it loads, see [here](https://github.com/Tribler/py-ipv8/blob/master/ipv8/REST/rest_manager.py#L77). `aiohttp`, unfortunately, does not allow adding more endpoints after the API has started. It would be helpful if IPv8 allows developers to customise the root endpoint before the API starts.

For now, I will keep using my custom `RESTManager` class, but I think there is some opportunity to get rid of all the `RESTManager` classes scattered around our projects. Any custom logic should be added by sub-classing the `RESTManager` in IPv8. What do we think about this?

TLDR: To reduce code, I want to re-use the `RESTManager` in IPv8 in applications that depend on IPv8 (e.g., AnyDex/Tribler) but I cannot customize the root endpoint.

**Comment by qstokkink** [comment]
I'd be fine with adding a `root_endpoint_class` argument to the `RESTManager.__init__`.

Usage would be something like this:

```python
class MyRootEndpoint(RootEndpoint):
    def setup_routes(self):
        # Alternatively: completely override ``setup_routes`` by not calling the super.
        super(MyRootEndpoint, self).setup_routes()
        self.add_endpoint('/alsothis', MyEndpoint)

rest_manager = RESTManager(ipv8_instance, MyRootEndpoint)
```

**Comment by devos50** [comment]
@qstokkink yes, that would give me the flexibility to use the endpoint I want 👍
